### DISCUSSION TITLE
Adventure map, right click info over garrison hero doesn't show the actual defending army

### DISCUSSION THREAD

**Discussion by Branikolog** [ihhub/fheroes2#8331] [General]
### Preliminary checks

- [X] I've checked that there aren't [**other open issues**](https://github.com/ihhub/fheroes2/issues?q=is%3Aissue) on the same topic.

### Describe the problem requiring a solution

With the recent improvement of the AI, that allows him to put some forces into castle garrison, checking info of the hero does not allow to see the actual army and troop arrangement.

![image](https://github.com/ihhub/fheroes2/assets/55348946/3bef4f11-7176-4d29-98e8-c6b0595116a1)

![image](https://github.com/ihhub/fheroes2/assets/55348946/570b2387-c351-426f-9de9-0c1e042c87b9)

![image](https://github.com/ihhub/fheroes2/assets/55348946/20e6ab24-c9b0-4d98-9dce-79e6ab7be83a)

![image](https://github.com/ihhub/fheroes2/assets/55348946/17e798a1-845e-43dc-b100-7cff954a2f40)

### Describe the possible solution

Right click info over the hero in a castle should display the actual forces that are going to meet you in the upcoming castle siege.

### Additional info

[Castle view.zip](https://github.com/ihhub/fheroes2/files/14012554/Castle.view.zip)

**Comment by oleg-derevenetz** [comment]
> Right click info over the hero in a castle should display the actual forces that are going to meet you in the upcoming castle siege.

How is that? If human player will do the same, you won't get this info, you will need to check both hero army and castle garrison (and you will need Thieves Guild somewhere to be able to do this). Why AI should be "special" in this regard? Also there is "Identify Hero" spell that offers you complete info about hero's army and parameters, but not about castle garrison, so putting part of the army to garrison is a legitimate mechanism to counter enemy intelligence, that's the core game mechanics. I don't think that we need to do something like this.

**Comment by Branikolog** [comment]
Hi, @oleg-derevenetz 

Well, in the original game AI always had his troops with a hero, while visiting any castle. So, while playing vs AI a player had always seen the troops and their arrangement he's going to be opposed during the siege.
Indeed, while opposing a human player you never know the arrangement of defending troops (in a case he put some forces into the garrison).
In other words now it is impossible to know, what troops arrangement you're going to meet in a siege, while in the OG you could.
So this is an improvement over the original game making it harder to capture castles.
If such representation issue was taken into account minding to make the game more challenging I actually have no objections and  am ready to close an issue.

But can we at least give a hand for novice or casual players and make such improvement a difficulty-restriced? :D
I mean allow such behaviour for all difficulties except easy and normal?

**Comment by zenseii** [comment]
> Hi, @oleg-derevenetz
> 
> Well, in the original game AI always had his troops with a hero, while visiting any castle. So, while playing vs AI a player had always seen the troops and their arrangement he's going to be opposed during the siege. Indeed, while opposing a human player you never know the arrangement of defending troops (in a case he put some forces into the garrison). In other words now it is impossible to know, what troops arrangement you're going to meet in a siege, while in the OG you could. So this is an improvement over the original game making it harder to capture castles. If such representation issue was taken into account minding to make the game more challenging I actually have no objections and am ready to close an issue.
> 
> But can we at least give a hand for novice or casual players and make such improvement a difficulty-restriced? :D I mean allow such behaviour for all difficulties except easy and normal?

Hi, @Branikolog. I specifically requested this change and gave the reasons why this is an improvement of the AI behavior. There is also the reason that the AI will get some extra movement points while being garrisoned like this. All in all, it's an overall improvement. As for having to limit this behavior, I think we need to playtest enough before proclaiming the end of the world for every AI improvement over the original AI. I've played several maps with this now and only found more enjoyment in seeing that the AI is not anymore doing the same as laying all your cards face up on the table in poker. This increases/restores the value of the thieves guild.

**Comment by kitovyj** [comment]
I don't think there is a reason to worry about AI being too challenging. As for now, it requires a special effort to lose a game on a balanced map even on impossible.

**Reply by Branikolog** [reply]
That depends on your skill. :)
Some time ago I showed this game to my friend's kid and looked, how he managed it.
We shouldn't forget that part of players are novice, casual and unexperienced players, so at least easy mode should behave like a true easy.
I believe we should keep adjusting the game difficulty levels to make the game more flexible regarding playability.

**Comment by Branikolog** [comment]
Hello, @zenseii 

I'm enjoying this improvement too (being an experienced player). 👍 
But since we're now on making the game more friendly to unexperienced players, limiting this trick to a higher difficulties could be one of those steps novice player would be faced on his road to the most challenging modes, how do you think?
For now I'm converting this into a discussion, since this implementation was intentional.

**Reply by zenseii** [reply]
> But since we're now on making the game more friendly to unexperienced players, limiting this trick to a higher difficulties could be one of those steps novice player would be faced on his road to the most challenging modes, how do you think?

Yes, this I very much agree with and I'm looking forward to having more time to discussing how the various difficulty levels should be made distinct.

**Comment by LeHerosInconnu** [comment]
Hello everyone,

I agree with @Branikolog's initial comment that the human player should always be able to know with a single click the composition of the opposing army in a possible battle located in a castle when there are troops in garrison and in the defending hero's army, regardless of whether there are more troops or not in total than the hero's army can hold (for example, four troops in garrison and three troops in the hero's army, i.e. a total of seven troops for five available slots in the hero's army).

As stated in this post: https://github.com/ihhub/fheroes2/issues/6755:

"...to allow the human player to see directly the troops that his hero will have to face, rather than having to make an estimation of the troops that will be present in the army of the defending AI hero in the castle."

A simple rule could also be applied to avoid teaching the AI to have to juggle garrisoned troops and those present in the hero's army to get maximum movement the next turn: https://github.com/ihhub/fheroes2/discussions/2979.

The aim of https://github.com/ihhub/fheroes2/issues/6755 and https://github.com/ihhub/fheroes2/discussions/2979 is for the AI to always have the best defending troops in the hero's army at the end of his turn (no "surprise" if the human player uses the dimension door spell).
In this way, the human player has all the information available with a simple right-click, and the AI doesn't have to worry about organizing its troops so that the hero has as much movement as possible the next turn (the best defending army and as much movement as possible not being achievable at the same time in most cases).
